community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/schema/ConstraintVerificationFailedKernelException.java

```diff
@@ -71,20 +71,26 @@ public String toString()
 
     private final Set<Evidence> evidence;
 
-    public ConstraintVerificationFailedKernelException( UniquenessConstraint constraint, Set<Evidence> evidence )
+    public ConstraintVerificationFailedKernelException( PropertyConstraint constraint, Set<Evidence> evidence )
     {
         super( Status.Schema.ConstraintVerificationFailure, "Existing data does not satisfy %s.", constraint );
         this.constraint = constraint;
         this.evidence = evidence;
     }
 
-    public ConstraintVerificationFailedKernelException( UniquenessConstraint constraint, Throwable failure )
+    public ConstraintVerificationFailedKernelException( PropertyConstraint constraint, Throwable failure )
     {
         super( Status.Schema.ConstraintVerificationFailure, failure, "Failed to verify constraint %s: %s", constraint,
                 failure.getMessage() );
         this.constraint = constraint;
         this.evidence = null;
     }
+    public ConstraintVerificationFailedKernelException( PropertyConstraint constraint )
```

Comment:
I'd vote for not having this constructor. It would be more helpful to see ids of nodes that violate `NOT NULL` constraint. Class `Evidence` should probably be made more generic to achieve this.